pkg/controller/extension/actuator.go

```diff
@@ -76,17 +76,17 @@ func (a *actuator) Reconcile(ctx context.Context, log logr.Logger, ext *extensio
 		return fmt.Errorf("error creating shoot client: %w", err)
 	}
 
-	if err := ReconcileShootInfoConfigMap(ctx, log, shootClient, config, cluster); err != nil {
-		return fmt.Errorf("error reconciling ConfigMap %q: %w", shootInfoConfigMapName, err)
-	}
-
 	if IsFluxBootstrapped(ext) {
 		log.V(1).Info("Flux installation has been bootstrapped already, will only reconcile secrets")
 
 		if err := ReconcileSecrets(ctx, log, a.client, shootClient, ext.Namespace, config, cluster.Shoot.Spec.Resources); err != nil {
 			return fmt.Errorf("error reconciling secrets: %w", err)
 		}
 
+		if err := ReconcileShootInfoConfigMap(ctx, log, shootClient, config, cluster); err != nil {
+			return fmt.Errorf("error reconciling ConfigMap %q: %w", shootInfoConfigMapName, err)
+		}
+
```

Comment:
I think this needs to be called outside the `if IsFluxBootstrapped(ext)` too (like `ReconcileSecrets`).

Otherwise if the configmap is referenced in the initial kustomization and it's not yet there the kustomization will not get ready and `SetFluxBootstrapped` will never be called.